guidelines/mwm-storage.md

```diff
@@ -1,3 +1,30 @@
-# Storage handling
+# Storage
 
+Input data, metadata & task output artifacts are all stored in a shared storage layer.
+
+### Architecture Diagram
+![store-arch.png](static/store-arch.png)
+
+### Storage Backend
+
+For the first releases (TBD) storage will be any S3 compatible resource. In a cloud-ready environment this can be AWS S3, and in environments restricted to on-premise storage it can be local services such as [Minio](http://min.io).
+
+In the future, alternative storage backends should be supported.
+
+### Access control
+
+#### Workflow Manager
+The MWM will be responsible for managing access to objects. It will have root level access and will be able to grant specific access to applications.
+
+#### Monai Informatics Gateway
+The informatics gateway will have write-only access to a specific storage bucket. This will be used to save the original input, and a reference will then be sent to the MWM.
+
+#### Workflow Manager Task Plugins
```

Comment:
Will the Workflow Manager or the Task Manager manage the access tokens for the plugins?
Readonly input and read-write for output?

Reply:
Since it's all running in the same process, and it's python (thus no private classes), the task plugins will technically be able to access all data. I think we're safe in trating plugins as trusted code, because one needs access to the source code in order to include a plugin anyway - in which case they could just as easily modify the source code...

That said - to prevent *accidental* issues, we should provide task plugins with an abstraction over the storage, which only provides access to the objects they need for the specific execution - and like you say, readonly for input.

Reply:
+1